FreeRTOS/FreeRTOS-Kernel
# Add Trace Hook Macros and function that returns the start of the stack.

Add Trace Hook Macros and function that returns the start of the stack.

The purpose of this pull request is to provide better support for tracing tools such as systemview. Some time ago I made a post on the FreeRTOS forum describing the current problem: https://forums.freertos.org/t/add-additional-tracing-functionality-to-kernel/16513 . This patchset adds a few new trace hooks that can be used by trace tools such as systemview. For now, I've only added support for the Cortex-M7 port, but if the change is a good improvement, I can add support for other ports. Note that this patch set is heavily inspired on the deprecated SystemView patch set.


Test Steps
----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
